src/coreclr/src/vm/tailcallhelp.cpp

```diff
@@ -503,6 +497,11 @@ MethodDesc* TailCallHelp::CreateCallTargetStub(const TailCallInfo& info)
     pCode->EmitLDC(info.ArgBufLayout.HasInstArg ? TAILCALLARGBUFFER_INSTARG_ONLY : TAILCALLARGBUFFER_ABANDONED);
     pCode->EmitSTIND_I();
 
+    // *pTailCallAwareRetAddr = NextCallReturnAddress();
```

Comment:
Why move it here? Assignments tends to cause stills. It is generally better to make assignments without stuff on the IL stack.

Reply:
I moved it close to the target call because now there were other calls in the IL stream between the `NextCallReturnAddress` and the actual call to the target. However it was still working properly because `Value` is force expanded, so the move is not strictly necessary. Code quality wise I checked that there is no difference (the JIT is already spilling because of the stind.i right above). However I don't have a strong opinion on where to put it.

Reply:
ok

Reply:
Actually, I would expect the original location to generate better code when there is a register pressure (`pTailCallAwareRetAddr` becomes dead earlier).

Reply:
Good point, I will move it back.